docs/source/guides/schema-design.md

```diff
@@ -0,0 +1,290 @@
+  updateUser(id: ID!, user: UserInput): User
+}
+
+input UserInput {
+  name: String
+  age: Int
+  address: String
+  phone: String
+}
+```
+
+<h3 id="mutation-responses">Responses</h3>
+
+Mutations have a higher chance of causing errors than queries since they are modifying data.  A common way to handle errors during a mutation is to simply `throw` an error.  While that's fine, throwing an error in the resolver will return an error to the caller and prevent a partial response, which could be useful in the event of a partial update.  Consider the following mutation example, which tries to update a user's `name` and `age`:
+
+```graphql
+mutation updateUser {
+  updateUser(id: 1, user: { age: -1, name: "Foo Bar" }){
+    name
+    age
+  }
+}
+```
+
+With validation in place, this mutation might cause an error since the `age` is a negative value.  While it’s possible that the entire operation should be stopped, there’s an opportunity to partially update the user’s record with the new `name` and return the updated record with the `age` left untouched.
+
+Luckily, the powerful structure of GraphQL mutations accommodates this use case and can return transactional information about the update alongside the records which have been changed which enables client-side updates to occur automatically.
+
+In order to provide consistency across a schema, we suggest introducing a `MutationResponse` interface which can be implemented on every mutation response in a schema and enables transactional information to be returned in addition to the normal mutation response object.
+
+A `MutationResponse` interface would look like this:
+
+```graphql
+interface MutationResponse {
+  code: String!
+  success: Boolean!
+  message: String!
+}
+```
+
+An implementing type would look like this:
+
+```graphql
+type UpdateUserMutationResponse implements MutationResponse {
+  code: String!
+  success: Boolean!
+  message: String!
+  user: User
+}
+```
+
+Calling a mutation that returns that `UpdateUserMutationResponse` type would result in a response that looks something like this:
+
+```json
+{
+  "data": {
+    "updateUser": {
+      "code": "200",
+      "success": true,
+      "message": "User was successfully updated",
+      "user": {
+        "id": "1",
+        "name": "Jane Doe",
+        "age": 35
+      }
+    }
+  }
+}
+```
+
+Let’s break this down, field by field:
+
+* `code` is a string representing a transactional value explaining details about the status of the data change. Think of this like an HTTP status code.
+* `success` is a boolean indicating whether the update was successful or not. This allows a coarse check by the client to know if there were failures.
+* `message` is a string that is meant to be a human-readable description of the status of the transaction. It is intended to be used in the UI of the product.
+* `post` is added by the implementing type `AddPostMutationResponse` to return back the newly created post for the client to use!
```

Comment:
This doesn't match the example above, which has `user`

Reply:
Addressed in 5c3d7b1ac4f068026a4c8c0976fb03b445013576.